src/controller/SetUpCodePairer.cpp

```diff
@@ -308,6 +308,11 @@ void SetUpCodePairer::NotifyCommissionableDeviceDiscovered(const Dnssd::Discover
     ConnectToDiscoveredDevice();
 }
 
+void SetUpCodePairer::CommissionerShuttingDown()
+{
+    ResetDiscoveryState();
```

Comment:
I don't think this actually works completely:

```
CHIP_ERROR BLEManagerImpl::CancelConnection()
{
    return CHIP_ERROR_NOT_IMPLEMENTED;
}
```

Our BLE operations on a lot of platforms aren't very cancelable.  For this to fully work, I think we'd either need those cancelable features in the platform BLE implementations, or we'd need some type of 'join' behavior from the commisisoner where it waits for pending BLE things.

Reply:
I agree this does not fully work as things stand.  The situation this is trying to resolve is when we have queued-up things discovered via dns-sd that we have not tried yet, so we don't start trying them.

If we have an active BLE discovery process, then as you note we don't have a great way to cancel it on some platforms...  We could try to block shutdown on it, I guess, but it's pretty difficult to do that sanely, espeially since by the time we get to here we're typically already shut down the Matter event loop.

I'd suggest we do followup work here in separate PRs.

Reply:
That makes sense.  If there's any concise way to have an issue filed that covers the 'in-flight' cases this PR doesn't cover, that would be welcome.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/20781 as an attempt at that.